Source/ARTRealtime.m

```diff
@@ -776,19 +778,15 @@ - (void)onConnected:(ARTProtocolMessage *)message {
         if ([message.connectionId isEqualToString:self.connection.id_nosync]) { // RTN15c6
             [self.logger debug:@"RT:%p connection \"%@\" has reconnected and resumed successfully", self, message.connectionId];
             // Reattach all channels with specified states
-            for (ARTRealtimeChannelInternal *channel in self.channels.nosyncIterable) {
-                if (channel.state == ARTRealtimeChannelAttaching || channel.state == ARTRealtimeChannelAttached || channel.state == ARTRealtimeChannelSuspended) {
-                    channel.state = ARTRealtimeChannelAttaching;
-                    [channel reattachWithReason:message.error callback:nil];
-                }
-            }
+            [self reattachChannelsOnResume:message];
             _resuming = false;
-        } else if (message.error) { //RTN15c7
+        } else if (message.error && [message.connectionId isEqualToString:self.connection.id_nosync]) { //RTN15c7
```

Comment:
`[message.connectionId isEqualToString:self.connection.id_nosync]` check was already made in the `if` above, so it will never reach this `else`.